src/nox_poetry/sessions.py

```diff
@@ -90,6 +93,38 @@ def _to_constraints() -> Iterator[str]:
     return "\n".join(_to_constraints())
 
 
+def to_index_url_args(
+    sources: List[Optional[Mapping[str, str]]]
+) -> Tuple[Optional[str]]:
+    """Convert the pip sources in to index url args."""
+    # TODO: add http basic-auth to the urls generated from the poetry config?
+    index_args = ()
```

Comment:
I would also like your thoughts on this.

Reply:
From the docs [here](https://python-poetry.org/docs/repositories/) and playing with it I think poetry uses both the user keychain and a fallback file to store those creds.. that could be a big lift to implement.